pkg/controllers/reconcilers.go

```diff
@@ -323,6 +326,16 @@ func (r *ChildReconciler) Reconcile(ctx context.Context, parent apis.Object) (ct
 	child, err := r.reconcile(ctx, parent)
 	if err != nil {
 		if apierrs.IsAlreadyExists(err) {
+			// check if the failed create is owned by the parent.
+			// the created child from a previous turn may be slow to appear in the informer cache, but shouldn't appear
+			// on the parent as being not ready.
+			apierr := err.(apierrs.APIStatus)
+			maybechild := r.ChildType.DeepCopyObject().(apis.Object)
+			_ = r.APIReader.Get(ctx, types.NamespacedName{Namespace: parent.GetNamespace(), Name: apierr.Status().Details.Name}, maybechild)
```

Comment:
This is how the controller-runtime clients work. An empty object is passed into the `.Get()` method by reference and the client populates it.

> Since we want to get the object which already existed when trying to create the child, why not use namespace r.ChildType.GetNamespace() and name r.ChildType.GetName() in the call to r.APIReader.Get()?

`ChildType` is empty, there is no namespace or name.

> I'm not sure why a deep copy of the child is used to prime maybechild. Don't we actually want to get the object which conflicted?

We don't want to mutate the `ChildType` which is shared with all turns of this reconciler.

> If the call to r.APIReader.Get() fails, then maybechild may not be updated. Wouldn't it then appear to be controlled by the parent, which would be misleading (if the conflicting, but (temporarily?) inaccessible, object isn't owned by the parent)?

We're already in an error handler that takes precedence. The lookup is an attempt to gracefully handle a corner case. If it fails, we don't detect that corder case and the mainline error handling continues.

Reply:
Ok, that's helpful. Thanks.